src/environment/environment.js

```diff
@@ -0,0 +1,127 @@
+/** @flow */
+import os from 'os';
+import { v4 } from 'uuid';
+import fs from 'fs-extra';
+import path from 'path';
+import npmInstall from '../utils/npm';
+import { loadScope } from '../scope';
+import { flattenDependencies } from '../scope/flatten-dependencies';
+import { BitId } from '../bit-id';
+import { Component } from '../consumer/component/consumer-component'
+import { BITS_DIRNAME } from '../constants';
+import { mergeAll } from 'ramda';
+
+const root = path.join(os.tmpdir(), 'bit');
+const currentPath = process.cwd();
+
+export default class Environment {
+  path: string;
+  component: Component;
+  componentsDependencies: ConsumerComponent[] = [];
+
+  constructor() {
+    this.path = path.join(root, v4());
+    console.log(this.path);
+  }
+
+  init(): Promise<*> {
+    return new Promise((resolve, reject) => {
+      fs.ensureDir(this.path, (err) => {
+        if (err) return reject(err);
+        return resolve();
+      });
+    });
+  }
+
+  importById(rawId: string): Promise<Object> {
+    // loadScope works for local scope and remote scope
+    return loadScope(currentPath).then((scope) => {
+      const bitId = BitId.parse(rawId, scope.name);
+      return scope.get(bitId)
+        .then((component) => {
+          component.component.write(this.path, true, true);
+          this.component = component.component;
+          return component;
+        });
+    });
+  }
+
+  writeToEnvironmentDir(componentDependencies: versionDependencies[]): Promise<Component[]> {
+    const componentsDir = path.join(this.path, BITS_DIRNAME);
+    const components = flattenDependencies(componentDependencies);
+
+    const bitDirForEnvironmentImport = (component: Component) => {
+      return path.join(
+        componentsDir,
+        component.box,
+        component.name,
+        component.scope,
+        component.version.toString(),
+      );
+    };
+
+    return Promise.all(components.map((component) => {
+      const bitPath = bitDirForEnvironmentImport(component);
+      return component.write(bitPath, true);
+    }));
+  }
+
+  importDependencies(): Promise<ConsumerComponent[]> {
+    if (!this.component) return Promise.reject();
```

Comment:
> The first issue I have found is that it doesn't install the compiler nor the tester. I have also tried to run it from the command line: bit import bit.utils/array/diff -e -s, but I got the same results. Am I missing something?

The syntax seems to be fine, it should work, let's debug it and see why it's not importing the devDependencies.

> Another issue is the performance. By using a consumer it performs unnecessary steps such as writing to .bit/objects.


That's not something special for the environment, a consumer should not create cache for imported components. when it'll be fixed, it will also be for the environment.

> Ignore the method names, they will be replaced.

After `init()` you have a consumer instance. I suggest keeping a reference to it on the environment state. so you could do environment.consumer.init() and just use the consumer as it's usually done.